src/pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -608,12 +610,12 @@ def __resolve_ckpt_dir(self, trainer: "pl.Trainer") -> None:
 
         self.dirpath = ckpt_path
 
-    def __warn_if_dir_not_empty(self, dirpath: _PATH) -> None:
-        if self.save_top_k != 0 and self._fs.isdir(dirpath) and len(self._fs.ls(dirpath)) > 0:
+    def __warn_if_dir_not_empty(self, dirpath: Optional[_PATH]) -> None:
```

Comment:
But dirpath can't be None, right? This type will never be satisfied. 
After the call `self.__resolve_ckpt_dir(trainer)` the dirpath is guaranteed to be not None.